src/common/post.ts

```diff
@@ -487,6 +487,23 @@ export const processApprovedModeratedPost = async (
 
   if (externalLink) {
     const cleanUrl = standardizeURL(externalLink);
+    const existingPost = await getExistingPost(con, cleanUrl);
+
+    if (existingPost) {
+      const post = await createSharePost({
+        con,
+        args: {
+          sourceId,
+          commentary: content,
+          authorId: createdById,
+          postId: existingPost.id,
+          visible: existingPost.visible,
+        },
+      });
+
+      return { ...moderated, postId: post.id };
+    }
```

Comment:
So this is just in case post already exists and now instead of creating it we add a shared post for approval?